### ISSUE TITLE
Pretty print debug logs

### ISSUE THREAD

**Issue by containscafeine** [kedgeproject/kedge#123]
Currently, the debug logs look like -

```
time="2017-07-06T12:28:53+05:30" level=debug msg="object unmrashalled: spec.App{Name:"", Replicas:(*int32)(nil), Labels:map[string]string(nil), PersistentVolumes:[]spec.PersistentVolume(nil), ConfigMaps:[]spec.ConfigMapMod(nil), Services:[]spec.ServiceSpecMod(nil), Ingresses:[]spec.IngressSpecMod(nil), Containers:[]spec.Container(nil), PodSpec:v1.PodSpec{Volumes:[]v1.Volume(nil), InitContainers:[]v1.Container(nil), Containers:[]v1.Container(nil), RestartPolicy:"", TerminationGracePeriodSeconds:(*int64)(nil), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"", NodeSelector:map[string]string(nil), ServiceAccountName:"", DeprecatedServiceAccount:"", NodeName:"", HostNetwork:false, HostPID:false, HostIPC:false, SecurityContext:(*v1.PodSecurityContext)(nil), ImagePullSecrets:[]v1.LocalObjectReference(nil), Hostname:"", Subdomain:""}, DeploymentSpec:v1beta1.DeploymentSpec{Replicas:(*int32)(nil), Selector:(*unversioned.LabelSelector)(nil), Template:v1.PodTemplateSpec{ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.PodSpec{Volumes:[]v1.Volume(nil), InitContainers:[]v1.Container(nil), Containers:[]v1.Container(nil), RestartPolicy:"", TerminationGracePeriodSeconds:(*int64)(nil), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"", NodeSelector:map[string]string(nil), ServiceAccountName:"", DeprecatedServiceAccount:"", NodeName:"", HostNetwork:false, HostPID:false, HostIPC:false, SecurityContext:(*v1.PodSecurityContext)(nil), ImagePullSecrets:[]v1.LocalObjectReference(nil), Hostname:"", Subdomain:""}}, Strategy:v1beta1.DeploymentStrategy{Type:"", RollingUpdate:(*v1beta1.RollingUpdateDeployment)(nil)}, MinReadySeconds:0, RevisionHistoryLimit:(*int32)(nil), Paused:false, RollbackTo:(*v1beta1.RollbackConfig)(nil), ProgressDeadlineSeconds:(*int32)(nil)}}
" 
time="2017-07-06T12:28:53+05:30" level=debug msg="object after population: &spec.App{Name:"", Replicas:(*int32)(nil), Labels:map[string]string{"app":""}, PersistentVolumes:[]spec.PersistentVolume(nil), ConfigMaps:[]spec.ConfigMapMod(nil), Services:[]spec.ServiceSpecMod(nil), Ingresses:[]spec.IngressSpecMod(nil), Containers:[]spec.Container(nil), PodSpec:v1.PodSpec{Volumes:[]v1.Volume(nil), InitContainers:[]v1.Container(nil), Containers:[]v1.Container(nil), RestartPolicy:"", TerminationGracePeriodSeconds:(*int64)(nil), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"", NodeSelector:map[string]string(nil), ServiceAccountName:"", DeprecatedServiceAccount:"", NodeName:"", HostNetwork:false, HostPID:false, HostIPC:false, SecurityContext:(*v1.PodSecurityContext)(nil), ImagePullSecrets:[]v1.LocalObjectReference(nil), Hostname:"", Subdomain:""}, DeploymentSpec:v1beta1.DeploymentSpec{Replicas:(*int32)(nil), Selector:(*unversioned.LabelSelector)(nil), Template:v1.PodTemplateSpec{ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.PodSpec{Volumes:[]v1.Volume(nil), InitContainers:[]v1.Container(nil), Containers:[]v1.Container(nil), RestartPolicy:"", TerminationGracePeriodSeconds:(*int64)(nil), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"", NodeSelector:map[string]string(nil), ServiceAccountName:"", DeprecatedServiceAccount:"", NodeName:"", HostNetwork:false, HostPID:false, HostIPC:false, SecurityContext:(*v1.PodSecurityContext)(nil), ImagePullSecrets:[]v1.LocalObjectReference(nil), Hostname:"", Subdomain:""}}, Strategy:v1beta1.DeploymentStrategy{Type:"", RollingUpdate:(*v1beta1.RollingUpdateDeployment)(nil)}, MinReadySeconds:0, RevisionHistoryLimit:(*int32)(nil), Paused:false, RollbackTo:(*v1beta1.RollbackConfig)(nil), ProgressDeadlineSeconds:(*int32)(nil)}}
" 
time="2017-07-06T12:28:53+05:30" level=debug msg="app: , deployment: <*>&Deployment{ObjectMeta:k8s_io_kubernetes_pkg_api_v1.ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: ,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,},Spec:DeploymentSpec{Replicas:nil,Selector:nil,Template:k8s_io_kubernetes_pkg_api_v1.PodTemplateSpec{ObjectMeta:ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{app: ,},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,},Spec:PodSpec{Volumes:[],Containers:[],RestartPolicy:,TerminationGracePeriodSeconds:nil,ActiveDeadlineSeconds:nil,DNSPolicy:,NodeSelector:map[string]string{},ServiceAccountName:,DeprecatedServiceAccount:,NodeName:,HostNetwork:false,HostPID:false,HostIPC:false,SecurityContext:nil,ImagePullSecrets:[],Hostname:,Subdomain:,},},Strategy:DeploymentStrategy{Type:,RollingUpdate:nil,},MinReadySeconds:0,RevisionHistoryLimit:nil,Paused:false,RollbackTo:nil,ProgressDeadlineSeconds:nil,},Status:DeploymentStatus{ObservedGeneration:0,Replicas:0,UpdatedReplicas:0,AvailableReplicas:0,UnavailableReplicas:0,Conditions:[],},}
```

It'd be a lot more "debuggable" if we pretty print these logs :)

**Comment by surajssd** [comment]
Yes pass it to `json.MarshalIndent(v, "", "  ")` or similar thing in yaml should do the magic

**Comment by surajssd** [comment]
@containscafeine can you elaborate on how would you like to see the output of the debug logs?

In short I mean define `prettyPrint` ?

**Comment by kadel** [comment]
How would you like to pretty print it? 
Why don't you like current format?

**Comment by kadel** [comment]
removed user-experience label, as debug log are not something that is for users, but it is meant to help kedge developers (us)

**Comment by containscafeine** [comment]
Now that we have `prettyPrintObjects()` function, I think we can reuse that to pretty print log messages.

**Comment by containscafeine** [comment]
Fixed in #596
